### PR TITLE
implement Nextauth

### PR DESCRIPTION
**Pull request by gcode-de**

### CODE REVIEW

File: pages/api/auth/[...nextauth].js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  ],
+
+  adapter: MongoDBAdapter(clientPromise),
+
+  pages: {
+    signIn: "/auth/signin",
+    signOut: "/",
+    // error: '/auth/error', // Error code passed in query string as ?error=
+    // verifyRequest: '/auth/verify-request', // (used for check email message)
+    // newUser: '/auth/new-user' // New users will be directed here on first sign in (leave the property out if not of interest)
```
Comment (NikRoe) [inline pages/api/auth/[...nextauth].js]: do you need those comments?

File: pages/auth/signin.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            </StyledButton>
+          </div>
+        ))}
+      </StyledWrapper>
+    </>
+  );
+}
+
+const getProviderIcon = (providerName) => {
+  switch (providerName) {
```
Comment (NikRoe) [inline pages/auth/signin.js:30]: this is a very long switch - case ;-)

File: pages/auth/signin.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    case "Google":
+      return <GoogleIcon className="provider-icon" />;
+    default:
+      return null;
+  }
+};
+
+export async function getServerSideProps(context) {
```
Comment (NikRoe) [inline pages/auth/signin.js:38]: 🥳

File: pages/profile/hasCooked.js [hunk 4]
```diff
@@ -22,20 +22,31 @@ export default function HasCooked({
 
   if (error) {
     <div>
-      <Header text={"schon gekocht 🥗"} />
+      <Header text={"Schon gekocht"} />
```
Comment (NikRoe) [inline pages/profile/hasCooked.js:25]:
```suggestion
      <Header text="Schon gekocht" />
```

File: pages/profile/hasCooked.js [hunk 5]
```diff
@@ -22,20 +22,31 @@ export default function HasCooked({
...
       </>
     );
   }
 
+  if (!hasCookedRecipes.length) {
+    return (
+      <>
+        <Header text="Schon gekocht" />
+        <StyledArticle>
+          <StyledUl>{`Du hast noch keine Rezepte als "gekocht" markiert.`}</StyledUl>
```
Comment (NikRoe) [inline pages/profile/hasCooked.js:51]:
```suggestion
          <StyledUl>Du hast noch keine Rezepte als "gekocht" markiert.</StyledUl>
```

File: pages/profile/myRecipes.js [hunk 6]
```diff
@@ -70,26 +70,30 @@ export default function MyRecipes({
...
-              <MealCard
-                key={recipe._id}
-                recipe={recipe}
-                isFavorite={getRecipeProperty(recipe._id, "isFavorite")}
-                onToggleIsFavorite={toggleIsFavorite}
-              ></MealCard>
-            );
-          })}
+          {myRecipes.length
+            ? myRecipes?.map((recipe) => {
```
Comment (NikRoe) [inline pages/profile/myRecipes.js:86]: do you need the second check?

File: pages/recipe/[id]/index.js [hunk 7]
```diff
@@ -190,6 +190,11 @@ export default function DetailPage({
               : "var(--color-lightgrey)"
           }
           onClick={() => {
+            console.log(user);
```
Comment (NikRoe) [inline pages/recipe/[id]/index.js]:
```suggestion
```

**Review by NikRoe — APPROVED** [review]
Hi there,

nicely done.

I left some questions / comments for you to take a look at.
Please see you need to change some part of your code after answering those questions.

In any case, PR is approved 👍

File: pages/profile/hasCooked.js (hunk 5 above)
Comment (gcode-de) [reply pages/profile/hasCooked.js:51]: Apparently this is needed as there are quotation marks within the text...

File: pages/auth/signin.js (hunk 2 above)
Comment (gcode-de) [reply pages/auth/signin.js:30]: built for scalability :-D

**Pull request merged**